analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ObjectsShouldNotBeDisposedMoreThanOnce.CSharp8.cs

```diff
@@ -163,3 +163,32 @@ public void Ternary(Disposable a, Disposable b, Disposable x, Disposable y, bool
         y.Dispose(); // Noncompliant
     }
 }
+
+// Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/8642
+public class Repro_8642
+{
+    async Task Method(InstanceDisposable instance)
+    {
+        StaticDisposable.Dispose();
+        StaticDisposable.Dispose();             // Compliant - none of these methods are coming from the System.IDisposable or System.IAsyncDisposable interfaces
+        await StaticDisposable.DisposeAsync();
+        await StaticDisposable.DisposeAsync();  // Compliant
+
+        instance.Dispose();
+        instance.Dispose();                     // Compliant
+        await instance.DisposeAsync();
+        await instance.DisposeAsync();          // Compliant
+    }
+
+    public static class StaticDisposable
+    {
+        public static void Dispose() { }
+        public static ValueTask DisposeAsync() => ValueTask.CompletedTask;
```

Comment:
The tests were added to the C# 8.0 test cases because `System.ValueTask` is not available in the .NET Framework and I didn't want to split it up. From the VB.NET test cases, I omitted the tests for `IAsyncDisposable`.